### PR TITLE
BUG: repr of np.datetime64('NaT') in Series/DataFrame with dtype object

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Discovered while investigating #12499

Before:

```
In [1]: pd.Series([np.datetime64('NaT')], dtype=object)
Out[1]:
0    NaN
dtype: object
```
After:

```
In [1]: pd.Series([np.datetime64('NaT')], dtype=object)
Out[1]:
0    NaT
dtype: object
```

### CODE REVIEW

**Review by TomAugspurger — APPROVED** [review]

File: pandas/io/formats/format.py [hunk 1]
```diff
@@ -946,7 +947,7 @@ def _format(x):
             if self.na_rep is not None and is_scalar(x) and isna(x):
                 if x is None:
                     return 'None'
-                elif x is NaT:
+                elif x is NaT or is_np_nat(x):
```
Comment (jreback) [inline pandas/io/formats/format.py]: use is_null_datetimelike instead of rolling your own

**Review by jreback — CHANGES_REQUESTED** [review]
see comment

File: pandas/io/formats/format.py (hunk 1 above)
Comment (mroeschke) [reply pandas/io/formats/format.py]: `is_null_datetimelike` is too permissive here (accepts `np.nan` and `None`)

Comment (jreback) [reply pandas/io/formats/format.py]: I don't want yet *another* NaT detection routine.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/formats/format.py (hunk 1 above)
Comment (mroeschke) [reply pandas/io/formats/format.py]: `is_null_datetimelike` doesn't _just_ detect NaT though. It just detects null values (None, np.nan, NaT, iNaT)

**Comment by jreback** [comment]
let's wait till we merge the min numpy 1.13 and then revist

File: pandas/io/formats/format.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 from pandas._libs import lib
 from pandas._libs.tslib import format_array_from_datetime
 from pandas._libs.tslibs import NaT, Timedelta, Timestamp, iNaT
+from pandas._libs.tslibs.nattype import is_np_nat
```
Comment (jreback) [inline pandas/io/formats/format.py]: this needs to be removed

File: pandas/io/formats/format.py [hunk 3]
```diff
@@ -941,10 +942,14 @@ def _format_strings(self):
 
         def _format(x):
             if self.na_rep is not None and is_scalar(x) and isna(x):
-                if x is None:
-                    return 'None'
-                elif x is NaT:
-                    return 'NaT'
+                try:
```
Comment (jreback) [inline pandas/io/formats/format.py]: can you add a comment here on what is going on here

File: pandas/tests/frame/test_repr_info.py [hunk 4]
```diff
@@ -512,3 +512,11 @@ def test_repr_categorical_dates_periods(self):
 
         df = DataFrame({'dt': Categorical(dt), 'p': Categorical(p)})
         assert repr(df) == exp
+
+    @pytest.mark.parametrize('arg', [np.datetime64, np.timedelta64])
+    @pytest.mark.parametrize('box, expected', [
+        [Series, '0    NaT\ndtype: object'],
+        [DataFrame, '     0\n0  NaT']])
+    def test_repr_np_nat_with_object(self, arg, box, expected):
+        result = repr(box([arg('NaT')], dtype=object))
```
Comment (jreback) [inline pandas/tests/frame/test_repr_info.py]: can you add the issue number

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by mroeschke** [comment]
Address comments and all green

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**